### PR TITLE
Feature/demo panel enhanced

### PR DESCRIPTION
**Pull request by fyffep**
This branch has my version of the merge conflict resolution

### CODE REVIEW

**Comment by tjsego** [comment]
@fyffep can you change the target to the develop branch?

**Comment by fyffep** [comment]
> @fyffep can you change the target to the develop branch?

@tjsego Rookie mistake. Thanks.

File: cc3d/player5/Plugins/ViewManagerPlugins/DemoBrowser.py [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+        demosPath = currPath.joinpath('CompuCell3D/Demos')
+        if demosPath.exists():
+            Configuration.setSetting("DemosPath", str(demosPath))
+            return demosPath
+
+    return None #No demos here!
+
+def getDemoList() -> list[Path]:
+    rootPath = getDemoRootPath()
+    globResults = rootPath.rglob('*.cc3d')
```
Comment (fyffep) [inline cc3d/player5/Plugins/ViewManagerPlugins/DemoBrowser.py:46]: I worry that `rglob` has a small chance of searching the user's entire file system if their directories are arranged a certain way. Is there a good way to limit the maximum search depth or cause a timeout?

**Comment by maciekswat** [comment]
After clicking Help->Open Demo on OSX I am getting this error:
```
qt.qpa.fonts: Populating font family aliases took 165 ms. Replace uses of missing font family "Courier" with one that exists to avoid this cost. 
Traceback (most recent call last):
  File "/Users/m/src/conda-build-repos/cc3d-player5/cc3d/player5/ViewManager/SimpleViewManager.py", line 687, in __open_demo_menu
    self.demo_browser = DemoBrowser(parent=self)
  File "/Users/m/src/conda-build-repos/cc3d-player5/cc3d/player5/Plugins/ViewManagerPlugins/DemoBrowser.py", line 269, in __init__
    self.searchIndex = createOrLoadSearchIndex()
  File "/Users/m/src/conda-build-repos/cc3d-player5/cc3d/player5/Plugins/ViewManagerPlugins/DemoBrowser.py", line 209, in createOrLoadSearchIndex
    idx = loadSearchIndex()
  File "/Users/m/src/conda-build-repos/cc3d-player5/cc3d/player5/Plugins/ViewManagerPlugins/DemoBrowser.py", line 183, in loadSearchIndex
    with open(indexFilePath, "r") as fp:
FileNotFoundError: [Errno 2] No such file or directory: '/Users/m/CompuCell3D/Demos/.demoindex'
```

**Comment by fyffep** [comment]
> It was the second error I got. The first one happened first time I used this branch and it had to do with encodings - I would not worry about it now because it might have been one-off error

Good thing you test my code! I still make simple mistakes sometimes. Please test again on the latest commit. I tried to fix both errors.

**Comment by maciekswat** [comment]
no worries. Will do that!

**Comment by maciekswat** [comment]
Added a few commits that fixed Player crashes when the user changed configuration while no simulation was running (not related to this PR at all) and also Added turning on and off Open Demo... option depending if the simulation is running or not

**Review by maciekswat — COMMENTED** [review]
looks good!

**Pull request merged**